### PR TITLE
Added option `extendModel` to the source model logic tree parser

### PR DESCRIPTION
**Pull request by micheles**
Closes https://github.com/gem/oq-engine/issues/5537 cleanly. Includes a test checking that
```xml
<?xml version="1.0" encoding="UTF-8"?>
<nrml xmlns:gml="http://www.opengis.net/gml"
      xmlns="http://openquake.org/xmlns/nrml/0.4">
  <logicTree logicTreeID="lt1">
    <logicTreeBranchSet uncertaintyType="sourceModel"
                        branchSetID="bs0">
      <logicTreeBranch branchID="b01">
        <uncertaintyModel>common1.xml</uncertaintyModel>
        <uncertaintyWeight>0.6</uncertaintyWeight>
      </logicTreeBranch>
      <logicTreeBranch branchID="b02">
        <uncertaintyModel>common2.xml</uncertaintyModel>
        <uncertaintyWeight>0.4</uncertaintyWeight>
      </logicTreeBranch>
    </logicTreeBranchSet>
    <logicTreeBranchSet uncertaintyType="extendModel"
                        branchSetID="bs1">
      <logicTreeBranch branchID="b11">
        <uncertaintyModel>extra1.xml</uncertaintyModel>
        <uncertaintyWeight>0.6</uncertaintyWeight>
      </logicTreeBranch>
      <logicTreeBranch branchID="b12">
        <uncertaintyModel>extra2.xml</uncertaintyModel>
        <uncertaintyWeight>0.4</uncertaintyWeight>
      </logicTreeBranch>
    </logicTreeBranchSet>
  </logicTree>
</nrml>
```
is equivalent to
```xml
<?xml version="1.0" encoding="UTF-8"?>
<nrml xmlns:gml="http://www.opengis.net/gml"
      xmlns="http://openquake.org/xmlns/nrml/0.4">
  <logicTree logicTreeID="lt1">
    <logicTreeBranchSet uncertaintyType="sourceModel"
                        branchSetID="bs">
      <logicTreeBranch branchID="b1">
        <uncertaintyModel>common1.xml extra1.xml</uncertaintyModel>
        <uncertaintyWeight>0.36</uncertaintyWeight>
      </logicTreeBranch>
      <logicTreeBranch branchID="b2">
        <uncertaintyModel>common1.xml extra2.xml</uncertaintyModel>
        <uncertaintyWeight>0.24</uncertaintyWeight>
      </logicTreeBranch>
      <logicTreeBranch branchID="b3">
        <uncertaintyModel>common2.xml extra1.xml</uncertaintyModel>
        <uncertaintyWeight>0.24</uncertaintyWeight>
      </logicTreeBranch>
      <logicTreeBranch branchID="b4">
        <uncertaintyModel>common2.xml extra2.xml</uncertaintyModel>
        <uncertaintyWeight>0.16</uncertaintyWeight>
      </logicTreeBranch>
    </logicTreeBranchSet>
  </logicTree>
</nrml>
```

### CODE REVIEW

File: openquake/commonlib/logictree.py [hunk 1]
```diff
@@ -1108,7 +1108,7 @@ def collect_source_model_data(self, branch_id, source_model):
         self.source_ids[branch_id].extend(ID_REGEX.findall(xml))
         self.source_types.update(SOURCE_TYPE_REGEX.findall(xml))
 
-    def apply_uncertainties(self, ltpath, sm):
+    def apply_uncertainties(self, ltpath, sm, converter):
```
Comment (mmpagani) [inline openquake/commonlib/logictree.py]: pls update the description of parameters. What's a 'converter'?

File: openquake/qa_tests_data/event_based/case_25/source_model_logic_tree2.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<nrml xmlns:gml="http://www.opengis.net/gml"
+      xmlns="http://openquake.org/xmlns/nrml/0.4">
+  <logicTree logicTreeID="lt1">
+    <logicTreeBranchSet uncertaintyType="sourceModel"
+                        branchSetID="bs0">
+      <logicTreeBranch branchID="b01">
+        <uncertaintyModel>common1.xml</uncertaintyModel>
+        <uncertaintyWeight>0.5</uncertaintyWeight>
```
Comment (mmpagani) [inline openquake/qa_tests_data/event_based/case_25/source_model_logic_tree2.xml]: I would use different weights here

Comment (micheles) [reply openquake/qa_tests_data/event_based/case_25/source_model_logic_tree2.xml]: Done

File: openquake/commonlib/logictree.py (hunk 1 above)
Comment (micheles) [reply openquake/commonlib/logictree.py]: Done

**Pull request merged**